dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue

```diff
@@ -140,30 +342,163 @@ export default defineComponent({
+      route: useRoute(),
+      waitingForData: false,
+      editMode: false,
+
+      postEuTaxonomyDataForFinancialsProcessed: false,
+      messageCount: 0,
+      postEuTaxonomyDataForFinancialsResponse: null as AxiosResponse<DataMetaInformation> | null,
+      humanizeString: humanizeString,
+      onThisPageLinksStart: [
+        { label: "Upload company reports", value: "uploadReports" },
+        { label: "Basic information", value: "basicInformation" },
+        { label: "Assurance", value: "assurance" },
+        { label: "Add KPIs", value: "addKpis" },
+      ],
+      onThisPageLinks: [] as { label: string; value: string }[],
+      kpisModel: [
+        { label: "Credit Institution", value: "creditInstitutionKpis" },
+        { label: "Investment Firm", value: "investmentFirmKpis" },
+        { label: "Insurance & Re-insurance", value: "insuranceKpis" },
+        { label: "Asset Management", value: "assetManagementKpis" },
+      ],
+      selectedKPIs: [] as { label: string; value: string }[],
+      confirmedSelectedKPIs: [] as { label: string; value: string }[],
+      computedFinancialServicesTypes: [] as string[],
+      reportingPeriodYear: new Date().getFullYear(),
+    };
+  },
+  computed: {
+    namesOfFilesToUpload(): string[] {
+      const namesFromFilesToUpload = this.filesToUpload.map((el) => el.name.split(".")[0]);
+      const namesFromUploadedFiles = this.uploadFiles.map((el) => el.name.split(".")[0]);
+      return [...new Set([...namesFromFilesToUpload, ...namesFromUploadedFiles])];
     },
-    inputClass: {
-      "formkit-input": false,
-      "p-inputtext": true,
-      "w-full": true,
+  },
+  watch: {
+    confirmedSelectedKPIs: function (newValue: { label: string; value: string }[]) {
+      this.computedFinancialServicesTypes = newValue.map((el: { label: string; value: string }): string => {
+        return this.euTaxonomyKPIsModel.companyTypeToEligibilityKpis[
+          el.value as keyof typeof this.euTaxonomyKPIsModel.companyTypeToEligibilityKpis
+        ];
+      });
     },
-    postEuTaxonomyDataForFinancialsProcessed: false,
-    messageCount: 0,
-    formInputsModel: {},
-    postEuTaxonomyDataForFinancialsResponse: null,
-    humanizeString: humanizeString,
-  }),
+    reportingPeriod: function (newValue: Date) {
+      this.reportingPeriodYear = newValue.getFullYear();
+    },
+  },
   props: {
     companyID: {
       type: String,
     },
   },
+  mounted() {
+    const dataId = this.route.query.templateDataId;
+
+    if (typeof dataId === "string" && dataId !== "") {
+      this.editMode = true;
+      void this.loadEuData(dataId);
+    }
+
+    this.onThisPageLinks = [...this.onThisPageLinksStart];
+  },
+
   methods: {
+    /**
+     * Loads the Dataset by the provided dataId and pre-configures the form to contain the data
+     * from the dataset
+     *
+     * @param dataId the id of the dataset to load
+     */
+    async loadEuData(dataId: string): Promise<void> {
```

Comment:
This method is really huge.
Is it possible to split it up and make it more reader-friendly?

Reply:
I will have to touch this method anyways when I will work on the new sub-tasks.

So you don't need to do the split up, since I can do it during my work on it.

Reply:
Note: Still too long. Modularize at the end!